### ISSUE TITLE
Record Wine Software 

### ISSUE THREAD

**Issue by Levan7** [MaartenBaert/ssr#283]
I was wondering is there a way to record software that is running throe wine ??

For example Delta force video games

**Comment by inukaze** [comment]
Someone have a Tutorial how to Record directly OpenGL, trough PlayOnLinux + Wine ???
For example if i wanna record my Gameplay of "Ragnarok Online" , i load the game from PlayOnLinux + Wine + Steam For Windows + Ragnarok Online . 

How i should do, for the Recording , works fine ???

**Comment by MaartenBaert** [comment]
Start the entire thing with ```ssr-glinject <command>```. It works even through WINE, unless PlayOnLinux is interfering somehow.

**Comment by inukaze** [comment]
Yes MaartenBaert : PlayOnLinux is interfering someone because i try various methods : 

Direct from PlayOnLinux and Launch Steam. Not Work

Try to launch manually : 
`bash "/opt/Videojuegos/Gestores/PlayOnPosix/4.2.10/playonlinux" --run "Steam" steam://rungameid/215100`

And after Record it, Dont work neither . 

i decided to make a miniscript with the follow content : 
```
#!/bin/bash
PATH="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/bin:$PATH"
export WINESERVER="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/bin/wineserver"
export WINELOADER="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/bin/wine"
export WINEDLLPATH="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/lib"
export WINEPREFIX="/home/inukaze/.PlayOnLinux/wineprefix/Steam"
export WINEDEBUG=-all
cd "/home/inukaze/.PlayOnLinux/wineprefix/Steam/drive_c/Program Files/Steam/steamapps/common/Ragnarok/"
wine Ragexe.exe -1rag1
```

and Ran from SSR Gui . using the Command Line : 
LD_PRELOAD=libssr-glinject.so SSR_GLINJECT_SHM=109051905 bash "/home/inukaze/Escritorio/SteamRagnarok"

When i press to Record . just create a file with 262 Bytes and not incremeted anymore.

I think i need some kind of Library for Wine, like a Wrapper to use libssr-glinject directly over wine
some idea came to me right now, "Maybe be the same thing with Steam Native for GNU/Linux" i must do :

```
#32BitsLib
cp /usr/lib/i386-linux-gnu/libssr-glinject.{la,so} $HOME/.local/share/Steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu

#64BitsLib 
cp /usr/lib/x86_64-linux-gnu/libssr-glinject.{la,so} $HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu
```

Or i can't use 
`ssr-glinject %command%`

In the "Parameters" game options for Record Directly OpenGL Steam Games.

But i dont know if with ".so" and ".la" works well i'll go to try it. if don't i guess i need a dll version adapted for Wine, or a "Direct3D" to "OpenGL" wrapper or something like that. xD

**Comment by MaartenBaert** [comment]
SSR_GLINJECT_SHM is from an old version of SSR, where did you even get that? The latest version doesn't have that.

Try this:
```
#!/bin/bash
PATH="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/bin:$PATH"
export WINESERVER="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/bin/wineserver"
export WINELOADER="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/bin/wine"
export WINEDLLPATH="/home/inukaze/.PlayOnLinux/wine/linux-x86/1.9.19/lib"
export WINEPREFIX="/home/inukaze/.PlayOnLinux/wineprefix/Steam"
export WINEDEBUG=-all
cd "/home/inukaze/.PlayOnLinux/wineprefix/Steam/drive_c/Program Files/Steam/steamapps/common/Ragnarok/"
ssr-glinject wine Ragexe.exe -1rag1
```
... with the latest SSR (0.3.7) of course. This will create the channel ```channel-<username>``` which is also the default in SSR.

**Comment by inukaze** [comment]
Hi there " MaartenBaert" , well i test anothers games like :

Warcraft 3 in OpenGL mode. using PlayOnLinux + Wine , and perfectly recorded
Warcraft 3 in Direct{X,3D} (I really dont remember what use this game) , and can be recorded pretty slow than OpenGL

Silent Hill 3 , can be recorded . too . after various test . i see really dont need it to copy the "so" and the "la" file to the wine lib folder. 

with "Enclave" in "OpenGL" mode and "Direct3D8" mode, i can record it too.

i can record SuperCrateBox too


the uniques i found problems to record it are : 
Cubetractor

and Ragnarok Online (The First)

ssr-inject wine Ragexe.exe -1rag1

and press Hotkeys for Record it. on the SSR Gui , i see the next message : 

[PageRecord :: StartOutput] Error: Could not get the size of the OpenGL application. Or the application has not played correctly, or the application has not yet created an OpenGL window. If you want to start recording before you start recording you must allow scaling and enter the video size manually.
[PageRecord :: StartOutput] Error: something has gone wrong during initialization.

Well i ask if because Ragnarok use "Direct3D7" ???

The Games with OpenGL and Direct3D8 and Direct3D9 , appers not have problems. but equal is something strange i can record some games.

**Comment by MaartenBaert** [comment]
This is an old 2D game, I suspect that it's not using 3D hardware acceleration. WINE will translate this to CPU-based rendering, not OpenGL, so that's why it doesn't work. Your GPU is not really being used by this game.

**Comment by inukaze** [comment]
Yes it is an old game 2.5D, it is a small 3D environment with 2D graphics superimposed. well in future version the SSR not have an option for Record "CPU-Based Rendering" direct from Game ??? or its better idea if i try to use another software to record it ??? (I accept suggestions) xD

**Comment by MaartenBaert** [comment]
You can already record it the normal way with SSR, without OpenGL injection.

There is no easy method to capture arbitrary 2D applications through injection, because there are a dozen different APIs that can be used. 3D is different, OpenGL is the only available API so it's easy to support.

**Comment by inukaze** [comment]
Feature, Wine options, is not has been added yet?

**Comment by MaartenBaert** [comment]
I'm not sure what you mean, SSR is not aware of Wine, it treats it like any other application.
